### PR TITLE
More docs

### PR DESCRIPTION
**Pull request by rvl**
- Add bors.md and nix.md from the cardano-sl repo.
- Update Nix configuration instructions for MacOS Mojave.
- Add two new gotcha docs.
- Rearrange gotchas into docs directory.

### CODE REVIEW

File: docs/nix.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+The `nix-daemon` must be restarted after editing `/etc/nix/nix.conf`
+for changes to take effect. Run `systemctl restart nix-daemon` on Linux
+or `sudo launchctl stop org.nixos.nix-daemon; sudo launchctl start
+org.nixos.nix-daemon` on macOS.
+
+We do not recommend using `~/.config/nix/nix.conf` unless you are a
+power user. It's simpler to have only one global config file.
+
+If using NixOS, see [`iohk-binary-cache.nix`](../nix/iohk-binary-cache.nix).
```
Comment (craigem) [inline docs/nix.md]: This link does not resolve, unless I'm missing something?

Comment (rvl) [reply docs/nix.md]: Fixed, thanks.

**Comment by rvl** [comment]
Thanxs for the review.

bors r+

**Comment by rvl** [comment]
Fixed master branch build.

bors r+

**Pull request merged**